community/values/src/main/java/org/neo4j/values/VirtualValue.java

```diff
@@ -69,4 +70,10 @@ public Boolean ternaryEquals( AnyValue other )
     public abstract VirtualValueGroup valueGroup();
 
     public abstract int compareTo( VirtualValue other, Comparator<AnyValue> comparator );
+
+    @Override
+    public long updateHash( HashFunction hashFunction, long hash )
+    {
+        return hashFunction.update( hash, hashCode() );
```

Comment:
If we ever need 64-bit hashing of non-storable values we need to remember to update this... WDYT about either just throwing `UnsupportedOperationException` here, or moving the `hashCode64()` and `updateHash` up into `Value` instead?